File: app/src/main/java/darkpoint/rubel/Fragments/AllCurrenciesFragment.java
package darkpoint.rubel.Fragments;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import darkpoint.rubel.APIFactory;
import darkpoint.rubel.Rate;
import darkpoint.rubel.CurrencyAdapter;
import darkpoint.rubel.R;
import darkpoint.rubel.ServerAPI;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class AllCurrenciesFragment extends Fragment {

    public static final String ARG_PAGE = "ARG_PAGE";
    int mPage;
    List<Rate> currencies = new ArrayList<>();

    public static AllCurrenciesFragment newInstance(int page)
    {
        Bundle args = new Bundle();
        args.putInt(ARG_PAGE, page);
        AllCurrenciesFragment fragment = new AllCurrenciesFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mPage = getArguments().getInt(ARG_PAGE);
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_all_currencies, container, false);

        final RecyclerView currency_list = view.findViewById(R.id.currencies_list);
        currency_list.addItemDecoration(new DividerItemDecoration(getContext(),
                DividerItemDecoration.VERTICAL));
        LinearLayoutManager llm = new LinearLayoutManager(getContext());
        currency_list.setLayoutManager(llm);

//        JSONGet jsonGet = new JSONGet("http://www.nbrb.by/API/ExRates/Rates?Periodicity=0");
//        currencies = jsonGet.getCurrencies();



        ServerAPI serverAPI = APIFactory.createApi();

        Call<List<Rate>> currenciesGSON = serverAPI.getAllRates(0);

        currenciesGSON.enqueue(new Callback<List<Rate>>() {
            @Override
            public void onResponse(Call<List<Rate>> call, Response<List<Rate>> response) {
                if(response.isSuccessful())
                {
                    CurrencyAdapter currencyAdapter = new CurrencyAdapter(response.body());
                    currency_list.setAdapter(currencyAdapter);
                    currencies = response.body();
                    Log.d("result", "onResponse: " + response.body().size());
                }
            }

            @Override
            public void onFailure(Call<List<Rate>> call, Throwable t) {
                Log.d("result", "onResponse: " + t.getLocalizedMessage());
            }
        });


//        CurrencyAdapter currencyAdapter = new CurrencyAdapter(currencies);
//        currency_list.setAdapter(currencyAdapter);
        return view;
    }
}

File: app/src/main/java/darkpoint/rubel/fragments/AllCurrenciesFragment.kt
package darkpoint.rubel.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import androidx.navigation.fragment.findNavController
import darkpoint.rubel.R
import darkpoint.rubel.activities.MainViewModel
import darkpoint.rubel.adapters.RateAdapter
import darkpoint.rubel.databinding.FragmentAllCurrenciesBinding

class AllCurrenciesFragment : Fragment() {

    private lateinit var binding: FragmentAllCurrenciesBinding
    private lateinit var viewModel: MainViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_all_currencies, container, false)

        binding.lifecycleOwner = this

        activity?.let {
            viewModel = ViewModelProviders.of(it).get(MainViewModel::class.java)
            binding.rateViewModel = viewModel
            //val currencyAdapter = CurrencyAdapter(viewModel.listCurrencies.value)
            binding.currenciesList.adapter = RateAdapter(RateAdapter.OnClickListener { rate ->
                val bundle = bundleOf("rateForDetail" to rate, "colorRate" to R.color.colorSelected)
                findNavController().navigate(R.id.action_global_detailFragment, bundle)
            })
        }

        viewModel.countCurrencies.observe(this, androidx.lifecycle.Observer {
            if (it == 4){
                binding.warningMessageAll.visibility = View.GONE
                binding.currenciesList.visibility = View.VISIBLE
            }
            else{
                binding.warningMessageAll.visibility = View.VISIBLE
                binding.currenciesList.visibility = View.GONE
            }
        })
        
        return binding.root
    }
}
